### ISSUE TITLE
add error handling

### ISSUE THREAD

**Issue by nurdtechie98** [nurdtechie98/drive-cli#2]
At several places only ideal cases have been considered. Need error handling at all the required places

**Comment by shikhag2018** [comment]
I would like to work on this issue

**Comment by inishchith** [comment]
@shikhag2018 Sure. Go Ahead

**Comment by shikhag2018** [comment]
How to get started with it.

**Comment by inishchith** [comment]
@shikhag2018 
- You can use `try except` for error and exception handling. [Read more](https://realpython.com/python-exceptions/)
- Use appropriate error / exception messages 

[Example](https://github.com/nurdtechie98/drive-cli/blob/a82cdc2c14d7208c65ca73f613b6c8ac040758c5/drive_cli/dcli.py#L224)

**Comment by nurdtechie98** [comment]
@shikhag2018 start with adding error handling for few functions where you feel there is a need of. We'll also let you know if we find other such places where you could do that in the mean time.

**Comment by shikhag2018** [comment]
how to run the code of drive-cli and know what exceptions are raised or i  have to manually read the code and think where execeptions can be raised.

**Comment by shivam1708** [comment]
Hi shikhag2018, I think manually reading the code can be tedious. It is better to try out things with an approach of finding a bug by giving buggy inputs and then handle the case. If it is pretty understandable by reading to include exception handling then you should. You can directly use drive-cli (npm install drive-cli) and test if you find bugs or follow the steps given (development mode) in the readme. Have you done that?

**Comment by shikhag2018** [comment]
yes i have done the setup in both development and usage mode  and i m using pip for python.

**Comment by shivam1708** [comment]
Sorry i wrote npm install drive-cli in flow. There is only python package. In "drive-cli/drive_cli" there is a file dcli.py which holds all the functions and utilities. In that file initially there are utility functions and from line 489 there are actual functions written for drive cli command line options. So try using different options on command line and understand the flow of execution. Once you understand that you can start testing with buggy inputs to find errors. If you have any difficulties feel free to ask :)

**Comment by Avni-Sharma** [comment]
PTAL #29

**Comment by Avni-Sharma** [comment]
I have added error handling for the first two functions namely login and drive data.
I also feel that a list of functions can be maintained with this issue here to which we can add error handling, as there are many people to contribute to and there are many places to add the error handling.

**Comment by nurdtechie98** [comment]
Agreed @Avni-Sharma , I have create checklist and we can start with separate contributions for each individual functionality.

**Comment by udit-001** [comment]
I would like to work on cat command!

**Comment by nurdtechie98** [comment]
sure @udit-001 go ahead

**Comment by udit-001** [comment]
@nurdtechie98 Done, please review!

**Comment by udit-001** [comment]
Can I work on log command, I noticed it requires error handling as well but it isn't mentioned in this thread?
